esp/esp/program/models/class_.py

```diff
@@ -1013,20 +1013,27 @@ def cancel(self, email_students=True, include_lottery_students=False, explanatio
         else:
             student_verbs = ['Enrolled']
 
+        email_ssis = email_students and all([sec.isCancelled() for sec in self.parent_class.get_sections() if sec!=self])
+
         context = {'sec': self, 'prog': self.parent_program, 'explanation': explanation}
         context['full_group_name'] = Tag.getTag('full_group_name') or '%s %s' % (settings.INSTITUTION_NAME, settings.ORGANIZATION_SHORT_NAME)
         context['site_url'] = Site.objects.get_current().domain
         context['email_students'] = email_students
         context['num_students'] = self.num_students(student_verbs)
         email_title = 'Class Cancellation at %s - Section %s' % (self.parent_program.niceName(), self.emailcode())
+        ssi_email_title = 'Class Cancellation at %s - Class %s' % (self.parent_program.niceName(), self.parent_class.emailcode())
         if email_students:
             email_content = render_to_string('email/class_cancellation.txt', context)
             template = Template(email_content)
             #   Send e-mail to each student
-            for student in self.students(student_verbs):
+            q_ssi = Q(studentsubjectinterest__subject=self.parent_class) & nest_Q(StudentSubjectInterest.is_valid_qobject(), 'studentsubjectinterest')
+            ssi_students = ESPUser.objects.filter(q_ssi).exclude(id__in=self.students(student_verbs)).distinct()
+            students = [(i,True) for i in self.students(student_verbs)] + [(i,False) for i in ssi_students]
```

Comment:
This seems like it will cause students with both SRs and SSIs to get two emails.